ports/mimxrt/mphalport.c

```diff
@@ -85,15 +85,11 @@ void mp_hal_stdout_tx_strn(const char *str, mp_uint_t len) {
             uint32_t n = len - i;
             uint32_t n2 = tud_cdc_write(str + i, n);
             if (n2 < n) {
-                while (!tud_cdc_write_flush()) {
-                    __WFI();
-                }
+                (void)tud_cdc_write_flush();
```

Comment:
This call needs to run in a loop until there is enough space to write.  Otherwise there is now out-going flow control and this loop can overflow the CDC.